File: Free Ice Cream.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    //freopen("input.txt","r",stdin);
    long long n,x,b,d=0;
    char a;
    cin >> n >> x;
    for (int i = 0 ; i < n ; ++i)
    {
        cin >> a >> b;
        if (a == '+')
            x += b;
        else
            if (b <= x)
                x -= b;
            else
                d++;
    }
    cout << x << " " << d;
    return 0;
}

File: Free Ice Cream.py
x, y = [int(x) for x in input().split()]
d = 0
for i in range(x):
    a, b = input().split()
    if a == '+':
        y += int(b)
    else:
        if int(b) <= y:
            y -= int(b)
        else:
            d += 1
print(y, d)
